src/test/java/uk/ac/ebi/ampt2d/metadata/MetadataApplicationTest.java

```diff
@@ -765,6 +770,20 @@ public void testAccessionValidation() throws Exception {
                 .andExpect(jsonPath("$..studies.length()").value(0));
     }
 
+    @Test
+    public void testSearchByAccessionVersionId() throws Exception {
+        String taxonomyUrl = postTestTaxonomy();
+        postTestStudy("EGAS0001", 1, "test_study", taxonomyUrl);
+        postTestFile("EGAF0001", 1);
+        postTestSample("EGAN0001", "Sample1");
+        mockMvc.perform(get("/studies?accessionVersionId=EGAS0001.1")).andExpect(status().isOk())
+                .andExpect(jsonPath("$..studies[0].accessionVersionId.accession").value("EGAS0001"));
+        mockMvc.perform(get("/files?accessionVersionId=EGAF0001.1")).andExpect(status().isOk())
+                .andExpect(jsonPath("$..studies[0].accessionVersionId.accession").value("EGAF0001"));
+        mockMvc.perform(get("/samples?accessionVersionId=EGAN0001.1")).andExpect(status().isOk())
+                .andExpect(jsonPath("$..studies[0].accessionVersionId.accession").value("EGAN0001"));
+    }
```

Comment:
A search by non-existing AccessionVersionId can be added to make it 360<sup>o</sup>; Optional

Reply:
ya it is tested in testAccessionValidation .